pkg/scheduler/core/generic_scheduler_test.go

```diff
@@ -2435,31 +2447,39 @@ func TestPreempt(t *testing.T) {
 			if len(node) == 0 && len(test.expectedNode) != 0 {
 				t.Errorf("expected node: %v, got: nothing", test.expectedNode)
 			}
-			if len(victims) != len(test.expectedPods) {
-				t.Errorf("expected %v pods, got %v.", len(test.expectedPods), len(victims))
+			if len(deletedPodNames) != len(test.expectedPods) {
+				t.Errorf("expected %v pods, got %v.", len(test.expectedPods), len(deletedPodNames))
 			}
-			for _, victim := range victims {
+			for victimName := range deletedPodNames {
 				found := false
 				for _, expPod := range test.expectedPods {
-					if expPod == victim.Name {
+					if expPod == victimName {
 						found = true
 						break
 					}
 				}
 				if !found {
-					t.Errorf("pod %v is not expected to be a victim.", victim.Name)
+					t.Errorf("pod %v is not expected to be a victim.", victimName)
 				}
-				// Mark the victims for deletion and record the preemptor's nominated node name.
-				now := metav1.Now()
-				victim.DeletionTimestamp = &now
 				test.pod.Status.NominatedNodeName = node
+				client.CoreV1().Pods(test.pod.Namespace).Update(context.TODO(), test.pod, metav1.UpdateOptions{})
```

Comment:
why are we doing this inside the loop iterating over the victims?

Reply:
As mentioned in your previous comment, we're now still fetching the pod from clientset, so the test needs to update the pod in fake clienset as well.

Reply:
sure, but why are we doing the update inside the loop iterating over ```deletedPodNames```, shouldn't the update happen once after or before the loop?

Reply:
Ah, the old logic which assigns "node" to "test.pod.Status.NominatedNodeName" misleads me. Updated.

PS: technically I can make `expectedPods` as a string set to shorten the code, but I have a PR on refactoring the whole TestPreempt testcase, which'd make this more neat. So for now, I'd prefer to just make minimum change.